File: leetcode/48. Rotate Image.cpp
class Solution {
public:
    void rotate(vector<vector<int>>& matrix) {
        int i = 0;
        int j = matrix.size() - 1;
        while(i < j){
            for(int m = 0; m < matrix[0].size(); ++m){
                int tmp = matrix[i][m];
                matrix[i][m] = matrix[j][m];
                matrix[j][m] = tmp;
            }
            ++i;
            --j;
        }
        
        for(int m = 0; m < matrix.size(); ++m){
            for(int n = 0; n < m; ++n){
                int tmp = matrix[m][n];
                matrix[m][n] = matrix[n][m];
                matrix[n][m] = tmp;
            }
        }
    }
};

File: leetcode/array/48. Rotate Image.py
class Solution:
    def rotate(self, matrix: List[List[int]]) -> None:
        """
        Do not return anything, modify matrix in-place instead.
        """
        n = len(matrix)
        for j in range(n):
            l = 0
            r = n-1
            while l < r:
                matrix[l][j], matrix[r][j] = matrix[r][j], matrix[l][j]
                l += 1
                r -= 1
        
        for i in range(n):
            for j in range(i):
                matrix[i][j], matrix[j][i] = matrix[j][i], matrix[i][j]
